src/main/java/codeu/controller/ParsingEngine.java

```diff
@@ -0,0 +1,33 @@
+package codeu.controller;
+
+import codeu.model.data.BotRequest;
+import codeu.model.data.BotStructure;
+import codeu.model.data.Message;
+import codeu.model.data.RoutingEngine;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+
+public class parsingEngine {
+ 	public void ParseCommands(Message message){
+ 		
+ 
+ 		final String commandRegex = "^/(?P<command_name>\\w+)(?:\\s+(?P<argument>\\w+))?\\s*$";
+ 		// compile the Regex command
+ 		Pattern pattern = Pattern.compile(commandRegex);
```

Comment:
If possible, I would make this a module-level constant as well, since recompiling regex on every request is expensive. Compiling it only when loaded into runtime is better for regex that does not change